assets/src/edit-story/components/richText/formatters/convert.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { Modifier } from 'draft-js';
+
+/**
+ * Internal dependencies
+ */
+import { ITALIC, UNDERLINE } from '../customConstants';
+import { weightToStyle } from './weight';
+
+function convertStyles(contentBlock, blockSelection, contentState) {
+  let updatedContentState = contentState;
+  let lastMetadata = null;
+  contentBlock.findStyleRanges(
+    (metadata) => {
+      lastMetadata = metadata;
+      return true;
+    },
+    (start, end) => {
+      // Get new list of styles for this range
+      const oldStyles = lastMetadata.getStyle().toArray();
+      const newStyles = getNewStyles(oldStyles);
+      // Create a selection for this range
+      const rangeSelection = blockSelection.merge({
+        anchorOffset: start,
+        focusOffset: end,
+      });
+      // Update content by removing all the old styles and applying all the new styles
+      oldStyles.forEach((oldStyle) => {
+        updatedContentState = Modifier.removeInlineStyle(
+          updatedContentState,
+          rangeSelection,
+          oldStyle
+        );
+      });
+      newStyles.forEach((newStyle) => {
```

Comment:
If the currently used font family does not have an option for a specific font-weight, should that font-weight not be applied? E.g. if the font family only supports regular but bold is pasted, it should stay regular, no?

Reply:
Hm, excellent point. But a complicated one. I'll check it out and see if it can easily be done without looking weird.

Reply:
This isn't easily accomplished. Also, there are various other ways to make parts of a text field bold even if the font family doesn't support it - by e.g. pressing the shortcut or the button for bold. So this is just another way of creating a "bold" text range without the font family actually supporting it.